server/modules/user/manager/userManager.js

```diff
@@ -98,6 +98,18 @@ const updateUserPrefs = async user => ({
   [User.keys.authGroups]: await AuthGroupRepository.fetchUserGroups(User.getUuid(user))
 })
 
+const acceptInvitation = async (user, name, client = db) =>
+  await client.tx(async t => {
+    const userUpdated = await UserRepository.updateUsername(user, name, t)
+    const groups = await AuthGroupRepository.fetchUserGroups(User.getUuid(user), t)
```

Comment:
when user accept an invitation, he has only one group associated to that survey.

Reply:
can it be that he is invited to more than one survey and when he logs in the first time he has multiple groups associated?

Reply:
no, the invitation happens only for one survey at the time. with the code you wrote, your are logging all groups, even for other surveys .